### PR TITLE
Limit Some Simultaneous Redis Requests

### PR DESCRIPTION
**Pull request by hiddenalpha**

### CODE REVIEW

**Review by Kusig — APPROVED** [review]

File: src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                        req.isDoneCalled = true;
+                        // give up lock because we don't know how much time mentor will use.
+                        req.lock.unlock();
+                        log.debug("call 'mentor.onDone()'");
+                        try {
+                            req.mentor.onDone(req.ctx);
+                        } finally {
+                            req.lock.lock(); // MUST get back our lock RIGHT NOW.
+                        }
+                    }else{
```
Comment (dominik-cnx) [inline src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java:90]: code format

File: src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.BiConsumer;
+
+import static java.lang.Thread.currentThread;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Level 1: KISS says do everything sequentially. Stop here! You're
```
Comment (dominik-cnx) [inline src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java:15]: Propose to say in the first sentence what it is, and then explain in detail. Now it seems the other way round.

**Review by dominik-cnx — APPROVED** [review]

File: src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java (hunk 2 above)
Comment (hiddenalpha) [reply src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java:25]: Oh yes :) Thx.

**Comment by mcweba** [comment]
Will you add this code (UpperBoundParallel) also for #170 ?

**Comment by hiddenalpha** [comment]
> Will you add this code (UpperBoundParallel) also for https://github.com/swisspost/vertx-redisques/issues/170 ?

Yes, working on it. But its quite hard to apply it properly there.

**Pull request merged**